src/NodaTime.Web/Models/TimerCache.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Threading;
+
+namespace NodaTime.Web.Models
+{
+    public class TimerCache<T>
+    {
+        private readonly Timer timer;
+        private readonly Func<T> provider;
+        private readonly ILogger logger;
+        private readonly object padlock = new object();
```

Comment:
(nit: first time I've seen `padlock` as the name of a lock variable; I would have expected just `lock` or `fooLock`.)